jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -117,28 +117,34 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
                 elif i == 0:
                     app.add_data_to_viewer("spectrum-viewer", data_label[i])
 
-        # reset display ranges, or build combined spectrum, when input is a SpectrumList instance
         if isinstance(data, SpectrumList):
-
-            # build combined spectrum
-            wlallarr = np.array(wlallorig)
-            fnuallarr = np.array(fnuallorig)
-            dfnuallarr = np.array(dfnuallorig)
-            srtind = np.argsort(wlallarr)
-            wlall = wlallarr[srtind]
-            fnuall = fnuallarr[srtind]
-            fnuallerr = dfnuallarr[srtind]
-
-            # units are not being handled properly yet.
-            unc = StdDevUncertainty(fnuallerr * flux_units)
-            spec = Spectrum1D(flux=fnuall * flux_units, spectral_axis=wlall * wave_units,
-                              uncertainty=unc)
-
-            # Make metadata layout conform with other viz.
-            spec.meta = standardize_metadata(spec.meta)
-
-            # needs perhaps a better way to label the combined spectrum
-            label = "Combined " + data_label[0]
-            app.add_data(spec, label)
-            if show_in_viewer:
-                app.add_data_to_viewer("spectrum-viewer", label)
+            if data[0].meta.get("INSTRUME") == "MIRI":
```

Comment:
I can see two use cases where this logic might fail:

* User provides a SpectrumList with mixed instruments and the first one happens to be MIRI and user does not want the composite spectrum.
* User provides a SpectrumList for some telescope we never heard of but user wants a composite spectrum.

Is there a way to generalize this solution to be less JWST specific?

Reply:
What about a new keyword so user can explicitly ask for composite spectrum or not. And if it is not given and detector is MIRI, then do the thing you do.

Reply:
@pllim Sorry it took me a while to come back to. I implemented your suggestion in [0fb4653](https://github.com/spacetelescope/jdaviz/pull/1659/commits/0fb46536b0df1dad08c9a1d26392f90791e991da).